python/mxnet/gluon/trainer.py

```diff
@@ -153,11 +161,55 @@ def set_learning_rate(self, lr):
         else:
             self._optimizer.set_learning_rate(lr)
 
-
     def step(self, batch_size, ignore_stale_grad=False):
         """Makes one step of parameter update. Should be called after
         `autograd.compute_gradient` and outside of `record()` scope.
 
+        Parameters
+        ----------
+        batch_size : int
+            Batch size of data processed. Gradient will be normalized by `1/batch_size`.
+            Set this to 1 if you normalized loss manually with `loss = mean(loss)`.
+        ignore_stale_grad : bool, optional, default=False
+            If true, ignores Parameters with stale gradient (gradient that has not
+            been updated by `backward` after last step) and skip update.
+        """
+        if not self._kv_initialized:
+            self._init_kvstore()
+
+        self._allreduce(batch_size, skip_pull=self._update_on_kvstore)
+        self.update(batch_size, ignore_stale_grad)
+
+    def allreduce(self, batch_size):
+        """For each parameter, reduce the gradients from different contexts. Should be called after
+        `autograd.compute_gradient`, outside of `record()` scope, and before `trainer.update`.
```

Comment:
what's compute_gradient?

Reply:
This comes from the API doc in step

Reply:
looks like its outdated. we should update